youtube_dl/extractor/brightcove.py

```diff
@@ -571,12 +571,28 @@ def build_format_id(kind):
         duration = float_or_none(json_data.get('duration'), 1000)
         if duration is not None and duration <= 0:
             is_live = True
+        common_res = ["160x90", "320x180", "480x270", "640x360", "768x432", "1024x576", "1280x720", "1366x768", "1920x1080"]
+        thumbnails = []
+        if '640x360' in json_data.get('poster'):
+            base_url = json_data.get('poster')
+            replace = '640x360'
+        else:
+            base_url = json_data.get('thumbnail')
+            replace = '160x90'
+        for res in common_res:
```

Comment:
With this:
```
        for n, res in enumerate(common_res):
```
you could replace `len(thumbnails)` by `n` as the value of `preference`.